parquet/src/file.rs

```diff
@@ -0,0 +1,64 @@
+use std::path::PathBuf;
+
+pub(crate) struct File {
+    file_data: Vec<u8>,
+    output_location: Location
+}
+
+impl File {
+    pub(crate) fn new(file_data: Vec<u8>, output_location: Location) -> File {
+        File {
+            file_data,
+            output_location,
+        }
+    }
+}
+
+pub(crate) enum Location {
+    DataWarehouse(PathBuf),
```

Comment:
I might be getting ahead since the PR is still a draft, or I might just be too noob with rust to properly understand what's going on😅 If so, call me out.

But should we make a DataWarehouse specific location? I feel like the sink should not know about the different destination types. Can we generalize it in some destination agnostic way?

If `process_substream` depends on a `trait FileSaver { save_file(&self, file: File) }` for example, that we can implement separately and handles the different storage types that a user might need. Implementations for `FileSaver` would live outside of the sink crate.

Reply:
So the sink ingests data from SF and is responsible for producing parquet files as an output. The only thing you're ever gonna need to do with these files is save them somewhere. So yes we could have an intermediatory stream that just spits out Vec<u8> to represent the file data and another service that then wraps them in this file struct, however I feel there is currently no need to do this as we don't need this intermediatory stream for any other purpose right now.